File: Python/small_truck.py
from Truck import Truck
from abc import ABC

class SmallTruck(Truck, ABC):
    MAX_VOLUME = 1000
    MAX_WEIGHT = 2000

    def add_pack(self, pack):
        if self.total_volume() + pack.get_volume() <= SmallTruck.MAX_VOLUME and \
                self.total_weight() + pack.get_weight() <= SmallTruck.MAX_WEIGHT:
            self.packages.append(pack)
        else:
            return False

    def total_volume(self):
        total = 0
        for pack in self.packages:
            total += pack.get_volume()
        return total

    def total_weight(self):
        total = 0
        for pack in self.packages:
            total += pack.get_weight()
        return total

File: C++/SmallTruck.cpp
#include "headers/SmallTruck.h"

SmallTruck::SmallTruck() {
    truckID = count;
}

int SmallTruck::getMaxWeight() const {
    return MAX_WEIGHT;
}

int SmallTruck::getMaxVolume() const {
    return MAX_VOLUME;
}

void SmallTruck::addPack(const Pack &p) {
    if (totalPacksVolume() + p.getVolume() <= MAX_VOLUME &&
        totalPacksWeight() + p.getWeight() <= MAX_WEIGHT) {
        packages.add(p);
    }
}

int SmallTruck::totalPacksVolume() {
    int total = 0;
    for (int i = 0; i < packages.size(); i++) {
        total += packages.get(i).getVolume();
    }
    return total;
}

int SmallTruck::totalPacksWeight() {
    int total = 0;
    for (int i = 0; i < packages.size(); i++) {
        total += packages.get(i).getWeight();
    }
    return total;
}
